src/main/java/org/terasology/pathfinding/model/Path.java

```diff
@@ -22,8 +22,33 @@
 /**
  * @author synopia
  */
-public class Path extends ArrayList<WalkableBlock> {
+public class Path {
     public static final Path INVALID = new Path();
+    private ArrayList<WalkableBlock> nodes = new ArrayList<>();
+
+    public void remove(int index) {
+        nodes.remove(index);
+    }
+
+    public void addAll(ArrayList<WalkableBlock> nodeGroup) {
```

Comment:
I think that this should be only a `List` or could even be just a `Collection`

Reply:
It can not be Collection, since those do not guarantee the maintenance of order - List does, though.

Reply:
This has been changed to a `List` instead of an `ArrayList` :+1: